java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/MySqlValidator.java

```diff
@@ -161,7 +161,33 @@ private void validateTableSchema() throws SQLException {
     }
 
     private void validatePrivileges() throws SQLException {
-        // TODO: validate MySQL user privileges
+        String[] privilegesRequired = {
+            "SELECT",
+            "RELOAD",
+            "SHOW DATABASES",
+            "REPLICATION SLAVE",
+            "REPLICATION CLIENT",
+            "LOCK TABLES"
+        };
+        try (var stmt = jdbcConnection.createStatement()) {
+            var res = stmt.executeQuery(ValidatorUtils.getSql("mysql.grants"));
+            while (res.next()) {
+                String grants = res.getString(1).toUpperCase();
+                // all privileges granted, check passed
+                if (grants.contains("ALL")) {
+                    break;
+                }
+                // check whether each privilege is granted
+                for (String privilege : privilegesRequired) {
+                    if (!grants.contains(privilege)) {
+                        throw ValidatorUtils.internalError(
```

Comment:
For validation errors we should use `invalid argument`, internal error is for other unexpected errors.